### ISSUE TITLE
Add a search button to the "Bookmarks" screen

### ISSUE THREAD

**Issue by ghost** [mozilla-mobile/fenix#12717]
Add a search function to the Bookmarks screen. Right now it is _really_ hard to manage bookmarks or find them. Let's say if I want to delete a bookmark then the only way to do so is to search that bookmark manually among hundreds of other bookmarks, this isn't practical! The only other way to do so is to type the bookmarks address in the Fenix's address bar then go to the bookmarked website and then tap the remove bookmark icon by tapping 3-dot menu. This is something that isn't practical if somebody wants to remove multiple bookmarks. And if someone wants to edit or move the bookmarks to different folders then it can only be done by manually finding them manually and moving them one-by-one. This also not practical.
Copying the bookmarked address' link is also something cannot be done easily without a search function.

These are all the reasons that make it really harder to manage, find and organize the bookmarks. And I think a search function is a must. This is already available in Chrome browser for example.

### Chrome browser's bookmarks search function: 
![Screenshot](https://user-images.githubusercontent.com/68282543/87878959-2ab6dc00-c9d7-11ea-9947-333bebf2579e.png)UNITO-UNDERSCORE!20200719-211658!
![Screenshot](https://user-images.githubusercontent.com/68282543/87878963-2f7b9000-c9d7-11ea-93a7-17c7c4e2b2ed.png)UNITO-UNDERSCORE!20200719-211711!

This will add to the history search function asked in #10046 but I can live with history search function not available for a moment as I already have the  suggestions appearing in address bar when I type something but for bookmarks it's just not about searching them but also about managing, editing, copying the bookmarked address' link, etc.

Thank you. :grinning:

**Comment by Poopooracoocoo** [comment]
would be great if it let you show it in its folder if it searches across folders.

**Comment by Abhishek8394** [comment]
I'd like to work on this. @sflorean is the screenshot the desired UX?

**Comment by eliserichards** [comment]
Hi @Abhishek8394! Since this is a feature request let's run it by the UX team to see if they want this or not 👍 If not, I'll find another issue you can work on :)

**Comment by Abhishek8394** [comment]
Just pinging to see if this issue is  still of interest or the review can be expedited. I miss this feature everyday 😿

**Comment by Mugurell** [comment]
@Abhishek8394 I support for searching bookmarks in https://github.com/mozilla-mobile/fenix/pull/23176 with a generic way - `configureSearchViewInMenu` to configure the search menu in any fragments. 
Seeing that you've already broken grounds on adding search support to when editing bookmarks I'll let it up to you if you want to continue your work from https://github.com/mozilla-mobile/fenix/pull/21585 rebased on these changes once they land.

**Comment by Abhishek8394** [comment]
I also need to be able to search for folder names, actually when adding a
bookmark, the common use case is to be able to find the folder. But I could
see searching a bookmark being useful if the user forgets the folder name,
in which case we'd like to show hierarchy info but not sure on UI for that.
Anyways I'll take a look, if the configurable search can fit these needs,
I'll use it. Thanks!

**Comment by Verdi** [comment]
@rocketsroger The bookmarks search results implementation in nightly includes the append arrow. We shouldn't have that. It can lead to weird situations where it looks like I can navigate to a url but I can't.

**Comment by rocketsroger** [comment]
> @rocketsroger The bookmarks search results implementation in nightly includes the append arrow. We shouldn't have that. It can lead to weird situations where it looks like I can navigate to a url but I can't.

Sure, created issue to track this https://github.com/mozilla-mobile/fenix/issues/24550

**Comment by Abhishek8394** [comment]
Hi, my original request was to have search functionality when **adding a bookmark.** The problems are:

-  When adding a bookmark, the user has to scroll all the way to a folder, and since the folders are not sorted in any particular order, it gets difficult. A search functionality on folder names would greatly help here.
- Or if you know a previous site/url you have added to that folder but not the folder name, in that case searching for it should show the folder info too? Not sure how many people would find this way of searching useful too.

I would be willing to add this and in fact had a related pull request which I would gladly modify to accommodate changes from [#23176](https://github.com/mozilla-mobile/fenix/pull/23176). 

Since this would be UX change, it would be great if someone can chime in and check if the team would like this feature.
